From 4f4cc5c79618e0a219a8fecf741df1b65103aed6 Mon Sep 17 00:00:00 2001
From: gbhakteX <gurudattax.bhakte@intel.com>
Date: Fri, 27 Jul 2012 16:39:05 +0530
Subject: [PORT FROM MAIN] Short noise occurs when seeking FLAC music contents.

BZ: 25416

The pcm data returned by the FLAC Decoder after seek operation is
causing noise. The pcm data/buffer is omitted from being sent back
to AudioPlayer.

Change-Id: I6d7f3b9a808b16f6d91d66f3f039187a90e30dbf
Orig-Change-Id: Icde67a5a9908bcd2e95633e3fc958b10b46ac91e
Signed-off-by: gbhakteX <gurudattax.bhakte@intel.com>
Reviewed-on: http://android.intel.com:8080/80760
Reviewed-by: M, Arulselvan <arulselvan.m@intel.com>
Tested-by: M, Arulselvan <arulselvan.m@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 media/libstagefright/FLACExtractor.cpp |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/media/libstagefright/FLACExtractor.cpp b/media/libstagefright/FLACExtractor.cpp
index 1fb58c1..d2257ef 100644
--- a/media/libstagefright/FLACExtractor.cpp
+++ b/media/libstagefright/FLACExtractor.cpp
@@ -662,6 +662,11 @@ MediaBuffer *FLACParser::readBuffer(bool doSeek, FLAC__uint64 sample)
     if (err != OK) {
         return NULL;
     }
+
+    if (doSeek) {
+        blocksize = 0;
+    }
+
     size_t bufferSize = blocksize * getChannels() * sizeof(short);
     CHECK(bufferSize <= mMaxBufferSize);
     short *data = (short *) buffer->data();
-- 
1.7.4.1

